src/download.cpp

```diff
@@ -206,6 +206,13 @@ void DownloadFile(const std::string& url, IDownloadSink *sink, int flags)
 
     if ( !filename_set )
     {
+        char ioptionurl[512];
```

Comment:
> but I'm happy to change it to whatever you prefer.

I’ll have you make the call — it’s obvious you have the choices well reasoned. However, *this* version of code for dynamically allocated buffer is buggy (it doesn’t check the first `InternetQueryOptionA` call’s result and it leaks memory b/c it never frees `optionurl`).

In any case, could you merge the final version into a single commit the describes the *why* (i.e. how it fixes the bug with failing to recognize redirected enclosure’s filename) rather than the *how* (which the code explains)? As it is now, it’s not clear from the commit message(s) alone *why* the changes are done (and the 25ee1d8 commit itself breaks compilation, breaking `git bisect`).

TIA!